src/octree/types.rs

```diff
@@ -81,3 +94,59 @@ pub struct Octree<T: Default + Clone + VoxelData, const DIM: usize = 1> {
     pub(in crate::octree) nodes: ObjectPool<NodeContent<T, DIM>>,
     pub(in crate::octree) node_children: Vec<NodeChildren<u32>>, // Children index values of each Node
 }
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Albedo {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+    pub a: u8,
+}
+
+impl Default for Albedo {
+    fn default() -> Self {
+        Self {
+            r: 0,
+            g: 0,
+            b: 0,
+            a: 0,
+        }
+    }
+}
+
+impl Albedo {
+    pub fn with_red(mut self, r: u8) -> Self {
+        self.r = r;
+        self
+    }
+
+    pub fn with_green(mut self, g: u8) -> Self {
+        self.g = g;
+        self
+    }
+
+    pub fn with_blue(mut self, b: u8) -> Self {
+        self.b = b;
+        self
+    }
+
+    pub fn with_alpha(mut self, a: u8) -> Self {
+        self.a = a;
+        self
+    }
+
+    pub fn is_transparent(&self) -> bool {
+        self.a == 0
+    }
+}
+
+#[test]
+fn albedo_size_is_4_bytes() {
+    const SIZE: usize = std::mem::size_of::<Albedo>();
+    const EXPECTED_SIZE: usize = 4;
+    assert_eq!(
+        SIZE, EXPECTED_SIZE,
+        "RGBA should be {} bytes wide but was {}",
+        EXPECTED_SIZE, SIZE
+    );
+}
```

Comment:
I am unsure this size restriction is necessary..

Reply:
To me it was more a safeguard, so we can be sure it's not suddenly 64bits by mistake and then wonder why we use double the amount of memory :D 

but I also see I have a typo in there, because it's not longer called RGBA but Albedo :D

Reply:
For the GPU part, the types `SizedNodes` stores the data. https://github.com/davids91/shocovox/blob/f72909fefaf1b6935a5ed38d8d383fdf65d50b03/src/octree/raytracing/types.rs#L32

It uses the outputs from the trait to do so: 
https://github.com/davids91/shocovox/blob/f72909fefaf1b6935a5ed38d8d383fdf65d50b03/src/octree/raytracing/classic_raytracing_on_bevy_wgpu.rs#L78

So by controlling the trait outputs we essentially control the size we Upload to GPU; The size restriction on `Albedo` is unrelated to this. I don't think such restriction need to apply on "CPU side" of the library.

SizedNode however has very strict size and alignm,ent restrictions through https://docs.rs/bevy/latest/bevy/render/render_resource/encase/trait.ShaderType.html#method.assert_uniform_compat

Reply:
So, I will remove it!